### PR TITLE
Deploy pim-api-docs with makefile commands

### PR DESCRIPTION
**Pull request by ahocquard**
In order to harmonize the deployment of the documentation across all the documentation:
- [x] use makefile to deploy
- [x] introduce environment variables to choose the server to deploy on
- [x] add these environment variables on circle CI 
- [x] use yarn.lock to be sure to have a consistent build
- [ ] deploy automatically master when a PR is merged

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,20 +2,16 @@ _DOCKER_IMAGE_TAG = pim-api-docs:latest
 _DOCKER_RUN = docker run -it --rm -u "$${UID}":"$${GID}" -v "$${PWD}":/opt/workdir -w /opt/workdir
 
 .PHONY: _docker-build _yarn-install serve deploy-staging
-.DEFAULT_GOAL := serve
+.DEFAULT_GOAL := build
 
 _docker-build:
```
Comment (jjanvier) [inline Makefile]: I would remove all the beginning `_` personnally

File: README.md [hunk 2]
```diff
@@ -6,54 +6,29 @@ The API documentation can be found here: [api.akeneo.com](http://api.akeneo.com)
...
+Your public SSH key should be deployed on the server (see Ansible configuration).
 
 ```bash
-make deploy-staging
+HOSTNAME=xxx PORT=xxx make deploy
 ```
 
+HOSTNAME is the server to deploy the documentation on.
+PORT is the SSH port to connect to the server.
+
```
Comment (jjanvier) [inline README.md]: To know the production and staging environments of api-docs, please read the [inventory](https://github.com/akeneo/ansible/blob/master/inventories/core.inventory).

Comment (ahocquard) [reply README.md]: As it's public, I don't know if we should precise it.  @BitOne  what do you think?

Comment (BitOne) [reply README.md]: The link will be forbidden for non Akeneo member, so it works for me.

**Pull request merged**